control-plane/api-gateway/cache/gateway.go

```diff
@@ -140,18 +136,5 @@ func (r *GatewayCache) subscribeToGateway(ctx context.Context, ref api.ResourceR
 		r.dataMutex.Lock()
 		r.data[common.NormalizeMeta(ref)] = derefed
 		r.dataMutex.Unlock()
-
-		event := event.GenericEvent{
-			Object: newConfigEntryObject(resource),
-		}
-
-		select {
-		case <-ctx.Done():
-			r.dataMutex.Lock()
-			delete(r.data, ref)
-			r.dataMutex.Unlock()
-			return
-		case r.events <- event:
-		}
```

Comment:
Solidifying my own understanding by writing it out:

Since the `r.events` channel is not buffered and nothing was reading from the channel, the first run through this code would block on the write. The context also was still open, so `ctx.Done()` wasn't happening either.

As a result, we would never move past this `select` to the next iteration of our `for` loop to refresh our cache, so it would very quickly become stale. Removing this code allows our loop to continue executing just as intended so that our cache is updated regularly.

Reply:
Commenting here so it's less likely to get lost. Removing the channel writing code seems fine to me, I believe it was originally there to re-trigger reconciliation if a derived Consul service was changed (like if someone manually deregistered the service for some odd reason, in which case we'd just re-register it), but clearly I forgot to hook it into our `Watch` routines 😬 

It's probably "more correct" to do it with the Watch, but that seems to be a small enough edge-case to me where I think dropping this here is fine. If we need to add it back in later though, we can.